src/ListBox/index.jsx

```diff
@@ -0,0 +1,121 @@
+import classNames from 'classnames';
+import { elementType, nonNegativeInteger, or } from 'airbnb-prop-types';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import ListBoxOption from '../ListBoxOption';
+import styles from './ListBox.scss';
+
+export default class ListBox extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.onFocus = this.onFocus.bind(this);
+    this.onKeyDown = this.onKeyDown.bind(this);
+
+    this.state = {
+      selectedOptionIndex: null,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { selectedOptionIndex } = nextProps;
+
+    if (this.shouldUpdateSelectedOptionIndex(selectedOptionIndex)) {
+      this.setState({
+        selectedOptionIndex,
+      });
+    }
+  }
+
+  onFocus() {
+    // if no ListBoxOption is selected, select first ListBoxOption on ListBox focus
```

Comment:
I guess this statement is mostly true. If `0` were selected, it would set the state to `0`, which is totally fine.